ClosedXML/Excel/XLWorksheet.cs

```
@@ -1350,35 +1336,57 @@ private void MoveNamedRangesColumns(XLRange range, int columnsShifted, IXLNamedR
 
         public void NotifyRangeShiftedRows(XLRange range, Int32 rowsShifted)
         {
-            if (RangeShiftedRows != null)
+            try
```

Why inside a `try`?

---

For a single failed `shift` would not leave events for entire sheet turned off. Exception is not swallowed though.

---

Oh, I see... no `catch` block.